src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Storage/FileSystemObsoletePluginsRemover.cs

```diff
@@ -107,10 +108,22 @@ private Task<bool> TryDeleteDirectory(string dir, CancellationToken cancellation
                     return true;
                 }
 
-                // First try to move the directory to the temp folder.
-                // If this succeeds, it means no other process was using any file
-                // in the directory, and we can safely delete it.
-                var toDelete = Path.Combine(Path.GetTempPath(), new DirectoryInfo(dir).Name + ".delete");
+                if (!TryDeleteContentsAtomic(dir))
+                {
+                    this.logger.Error($"Failed to delete {dir} because it is in use.");
+                    return false;
+                }
+
+
+                /*
+                 * The directory is not in use, and the plugins system won't return that it is available
```

Comment:
Shouldn't all files under the directory have been already deleted if they can at this point? I wonder if we still need to call `Directory.Move` now that we know the success doesn't guarantee the directory not in use? Can we just delete the current directory?

Reply:
Not necessarily true. There may have been a new file added to the directory after it was enumerated.

Reply:
I updated this code path a bit. Now, first we'll just attempt to delete the directory like you suggest. If that fails, we'll then attempt to rename the directory.

Reply:
Note that I also changed the move to just rename the folder within the same parent folder, instead of the temp directory. The reasoning here is
1. You can't move a directory across volumes. If the plugin is installed on a separate drive than the temp directory, the move will fail
2. If we move within the same folder, the next time cleanup code runs it'll attempt to delete it again.